src/node_watcher.js

```diff
@@ -270,16 +270,16 @@ NodeWatcher.prototype.processChange = function(dir, event, file) {
   var fullPath = path.join(dir, file);
   var relativePath = path.join(path.relative(this.root, dir), file);
   fs.lstat(fullPath, function(error, stat) {
+    var registered = this.registered(fullPath);
     if (error && error.code !== 'ENOENT') {
       this.emit('error', error);
     } else if (!error && stat.isDirectory()) {
       // win32 emits usless change events on dirs.
-      if (event !== 'change') {
+      if (event !== 'change' && registered) {
```

Comment:
I wrote something they may be of help(that wraps minimatch), it allows for:

1. matching against a collection of globs
2. checking to see if a given path could possibly be matched.

relevant code: https://github.com/stefanpenner/matcher-collection/blob/master/index.js#L23-L36